### PR TITLE
feat(core): password sign in with phone or email

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Add 2 APIs:

1. `/session/sign-in/password/email`
2. `/session/sign-in/password/sms`

Integration tests are not implemented yet 

## Testing

UTs.

### CODE REVIEW

File: packages/core/src/routes/session/utils.test.ts [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+    interactionDetails.mockResolvedValueOnce({ params: {} });
+    await expect(
+      signInWithPassword(createContext(), createProvider(), {
+        identifier: SignInIdentifier.Username,
+        password: 'password',
+        findUser: jest.fn(async () => mockUser),
+        logType: 'SignInUsernamePassword',
+        logPayload: { username: 'username' },
+      })
+    ).rejects.toThrowError();
```
Comment (darcyYe) [inline packages/core/src/routes/session/utils.test.ts:177]: nit: check error by error code could be more accurate?

File: packages/core/src/routes/session/password.test.ts [hunk 2]
```diff
@@ -116,60 +99,52 @@ describe('session -> password routes', () => {
...
+  });
 
-    it('throw if user found but wrong password', async () => {
-      interactionDetails.mockResolvedValueOnce({ params: {} });
-      const response = await sessionRequest.post(`${signInRoute}/username`).send({
-        username: 'username',
-        password: '_password',
-      });
-      expect(response.statusCode).toEqual(400);
+  it('POST /session/sign-in/password/email', async () => {
```
Comment (darcyYe) [inline packages/core/src/routes/session/password.test.ts:118]: nit: do we also need to check the case of inputting wrong password?

**Review by darcyYe — APPROVED** [review]
Overall LGTM

File: packages/core/src/routes/session/password.test.ts (hunk 2 above)
Comment (wangsijie) [reply packages/core/src/routes/session/password.test.ts:118]: It is already covered in line 142

File: packages/core/src/routes/session/utils.test.ts (hunk 1 above)
Comment (wangsijie) [reply packages/core/src/routes/session/utils.test.ts]: updated

**Pull request merged**